bundles/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/DSMRGenericBindingProvider.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2010-2014, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dsmr.internal;
+
+import org.openhab.binding.dsmr.DSMRBindingProvider;
+import org.openhab.core.binding.BindingConfig;
+import org.openhab.core.items.Item;
+import org.openhab.core.library.items.DateTimeItem;
+import org.openhab.core.library.items.NumberItem;
+import org.openhab.core.library.items.StringItem;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+
+/**
+ * This class is responsible for parsing the binding configuration.
+ * 
+ * @author M.Volaart
+ * @since 1.7.0
+ */
+public class DSMRGenericBindingProvider extends AbstractGenericBindingProvider
+		implements DSMRBindingProvider {
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public String getBindingType() {
+		return "dsmr";
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	public void validateItemType(Item item, String bindingConfig)
+			throws BindingConfigParseException {
+		// TODO: More advanced checking based on bindingConfig is possible
+		if (!(item instanceof NumberItem) && !(item instanceof StringItem) && !(item instanceof DateTimeItem)) {
+			throw new BindingConfigParseException(
+					"item '"
+							+ item.getName()
+							+ "' is of type '"
+							+ item.getClass().getSimpleName()
+							+ "', only NumberItems, DateTimeItems or StringItems are allowed - please check your *.items configuration");
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void processBindingConfiguration(String context, Item item,
+			String bindingConfig) throws BindingConfigParseException {
+		super.processBindingConfiguration(context, item, bindingConfig);
+		DSMRBindingConfig config = new DSMRBindingConfig();
+
+		config.bindingItem = bindingConfig.trim();
+
+		addBindingConfig(item, config);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String getItem(String itemName) {
```

Comment:
see above, please rename this method since it doesn't return an "Item" but something else